core/src/main/scala/tapir/SchemaFor.scala

```diff
@@ -53,4 +54,6 @@ object SchemaFor extends SchemaForMagnoliaDerivation {
   implicit def schemaForPart[T: SchemaFor]: SchemaFor[Part[T]] = new SchemaFor[Part[T]] {
     override def schema: Schema = implicitly[SchemaFor[T]].schema
   }
+
+  def oneOf[E, V](extractor: E => V, asString: V => String)(mapping: (V, SchemaFor[_])*): Discriminator = macro oneOfMacro[E, V]
```

Comment:
maybe this can return a `Schema.SCoproduct`? We have all the schemas, so it should just be a matter of creating the appropriate instance?